tasks/main.yml

```
@@ -17,6 +17,15 @@
     use: "{{ (__bootloader_is_ostree | d(false)) |
       ternary('ansible.posix.rhel_rpm_ostree', omit) }}"
 
+# Due to this bug, on latest grubby default kernel can change unexpectedly:
+# https://bugzilla.redhat.com/show_bug.cgi?id=2361624
+# So we need to get the default kernel in the beginning of the role and reset it at the end
+- name: Get default kernel in the beginning of the role
+  command: grubby --default-kernel
+  register: __bootloader_default_kernel
+  changed_when: false
+  when: bootloader_settings | selectattr('default', 'defined') | selectattr('default', 'equalto', true) | list | length == 0
```

```suggestion
  when: bootloader_settings | selectattr('default', 'defined') | selectattr('default') | list | length == 0
```
1. `equalto`, `eq`, and `==` are not supported on el7 - you have to use `match` with a regex instead, but 
2. you don't need to have a test here because the default test for `selectattr` (and `select`) is to evaluate the specified value in a boolean context i.e. it is the same as `selectattr('default', 'is', true)`